pipeline-scripts/release.groovy

```diff
@@ -21,24 +23,61 @@ def stageValidation(quay_url, name, advisory) {
         error("Payload ${name} already exists! Cannot continue.")
     }
 
-    // AMH - This may be optional?
-    if (advisory != "") {
-        echo "Verifying advisory exists"
-        res = commonlib.shell(
+    if (!advisory) {
+        echo "Getting current advisory for OCP $version from build data..."
+        res = withEnv(["_VERSION=$version"]) {
+            commonlib.shell(
                 returnAll: true,
-                script: "elliott get ${advisory}"
-        )
+                script: 'elliott --group=openshift-"${_VERSION}" get --json - --use-default-advisory image'
+            )
+        }
+        if(res.returnStatus != 0) {
+            error("🚫 Advisory number for OCP $version couldn't be found from ocp_build_data.")
+        }
+    } else {
+        echo "Verifying advisory ${advisory} exists"
+        res = withEnv(["_VERSION=$version", "_ADVISORY=$advisory"]) {
+            commonlib.shell(
+                returnAll: true,
+                script: 'elliott --group=openshift-"${_VERSION}" get --json - -- "${_ADVISORY}"'
+            )
+        }
 
         if(res.returnStatus != 0){
             error("Advisory ${advisory} does not exist! Cannot continue.")
         }
     }
+
+    def advisoryInfo = readJSON text: res.stdout
+    retval.advisoryInfo = advisoryInfo
+
+    echo "Verifying advisory ${advisoryInfo.id} (https://errata.engineering.redhat.com/advisory/${advisoryInfo.id}) status"
+    if (advisoryInfo.status != 'QE') {
+        error("🚫 Advisory ${advisoryInfo.id} is not in QE state.")
+    }
+    echo "✅ Advisory ${advisoryInfo.id} is in QE state."
+
+    // Extract live ID (if exists) from advisory info
+    // FIXME: Not sure if this is a stable way.
+    // If the advisory has a live ID assigned, let's say RHSA-2019:2594-03,
+    // advisoryInfo.errata_id should be 2594 and advisoryInfo.fulladvisory be RHSA-2019:2594-03
+    // while advisoryInfo.id remains the old internal advisory ID.
+    if (advisoryInfo.errata_id != advisoryInfo.id && advisoryInfo.fulladvisory && advisoryInfo.old_advisory) {
+        retval.liveID = (advisoryInfo.fulladvisory =~ /^(RH(?:B|S)A-\d+:\d+)-\d+$/)[0][1] // remove "-XX" suffix
```

Comment:
slightly simpler...
```suggestion
        retval.liveID = (advisoryInfo.fulladvisory =~ /^(RH[EBS]A-\d+:\d+)-\d+$/)[0][1] // remove "-XX" suffix
```
(We haven't used an RHEA in the past for $reasons but I hate to restrict things arbitrarily...)

Reply:
Good suggestion! 👍